### PR TITLE
Update semgrep pattern feature documentation in simple.md

### LINKED ISSUE #380: sgrep-lint API reference
To make sgrep-lint readily usable by security engineers, the sgrep-lint API should be fully documented.

This should include:

- How to invoke sgrep-lint, as a binary, and as a Docker image
- YAML structure
- Documentation on each key within the YAML schema

### PR DESCRIPTION
**Pull request by mschwager**
Fix #380. This change updates simple.md documentation with the latest
features and reorganizes the content for easier digestion.

See this link for the rendered Markdown: https://github.com/returntocorp/semgrep/blob/e2f60c3ddc8bd6d4e384429983357abe7043f7c3/docs/config/simple.md

### CODE REVIEW

File: docs/config/simple.md [hunk 1]
```diff
@@ -1,208 +1,344 @@
...
+```python
+from foo.bar import baz
+baz.qux()
 ```
 
-Because in Python there is usually no terminator \(e.g., `;`\), there is an ambiguity about `$Y` in the above, which could match a statement and also an expression that is then matched later.
+#### Constants
 
-### Match on import types
+JavaScript code will match propagated constants:
```
Comment (ievans) [inline docs/config/simple.md]: Golang works too. Maybe say "some languages" to make it more generic? not all languages have constants

**Review by ievans — APPROVED** [review]
Huge improvements!

**Review by minusworld — APPROVED** [review]
Looked it over; looks solid!

File: docs/config/simple.md (hunk 1 above)
Comment (mschwager) [reply docs/config/simple.md]: Good call, I'll add a Golang example too

**Pull request merged**